react exercise/users-information/src/pages/userDetails/UserDetails.tsx

```diff
@@ -0,0 +1,107 @@
+import { useState } from "react";
+import Toggle from "../../components/toggle/Toggle.tsx";
+import Search from "../../components/search/Search";
+import Topper from "../../components/topper/Topper.tsx";
+import CreatorCard from "../../components/creator-card/CreatorCard.tsx";
+import UserTable from "../../components/user-table/UserTable.tsx";
+import { User } from "../../types/user.ts";
+import { FilterButtons } from "../../components/filter-buttons/FilterButtons.tsx";
+import "./UserDetails.scss";
+
+export enum FilterType {
+  Cities = "cities",
+  Age = "age",
+}
+
+interface UserDetailsProps {
+  users: User[];
+}
+
+const filterUsers = (
+  users: User[],
+  activeFilter: FilterType | null,
+  searchQuery: string
+): User[] => {
+  return users.filter((user) => {
+    if (activeFilter === FilterType.Cities && searchQuery) {
+      return user.address.city
+        .toLowerCase()
+        .includes(searchQuery.toLowerCase());
+    }
+    if (activeFilter === FilterType.Age && searchQuery) {
+      return user.age.toString().startsWith(searchQuery);
+    }
+    return true; // If no filters are active, return all users
+  });
+};
+
+function UserDetails({ users }: UserDetailsProps) {
```

Comment:
I think the name of the component is a bit misleading, since this page is suppose to present multiple users and not a single one. Change the name to "UsersPage" or something like that ?